### DISCUSSION TITLE
No way to comment acl.conf on Kubernetes ?

### DISCUSSION THREAD

**Discussion by nishant-vvdn** [emqx/emqx#4566] [Q&A] [answered]
As it mentioned in official EMQX Docs: 

> The built-in ACL has the lowest priority and can be overridden by the ACL plugin. If you want to **disable** it, **you can comment all the rules**. After the rules file is changed, EMQ X Broker needs to be restarted to make them taking effect.

But there is not way to **comment all the rules in acl.conf** during **kubernetes deployment**. Kubernetes Deployment file (.yaml) does not have any way to do that.

**Comment by zmstone** [comment, marked as answer]
Hi @nishant-vvdn 
I believe this is not an issue. So I have decided to convert it to a discussion.
If there is indeed something wrong with our code, we can create an issue to follow up.

Could you let us know how you deploy emqx to k8s?
if it's using our helm chart:
you can edit this section in the values.yaml
```
emqxAclConfig: >
  {allow, {user, "dashboard"}, subscribe, ["$SYS/#"]}.
  {allow, {ipaddr, "127.0.0.1"}, pubsub, ["$SYS/#", "#"]}.
  {deny, all, subscribe, ["$SYS/#", {eq, "#"}]}.
  {allow, all}.
```

to add comments:

```
emqxAclConfig: >
  %% I want to disable 'dashboard' user.
  %% {allow, {user, "dashboard"}, subscribe, ["$SYS/#"]}.
  {allow, {ipaddr, "127.0.0.1"}, pubsub, ["$SYS/#", "#"]}.
  {deny, all, subscribe, ["$SYS/#", {eq, "#"}]}.
  {allow, all}.
```

**Reply by nishant-vvdn** [reply]
Hi, @zmstone.

I am not using helm chart to deploy EMQX. I'm using custom configuration so I'm using deployment file(.yaml) to deploy EMQX on K8s.

**Reply by zmstone** [reply]
I guess you can still generate acl.conf content from a template (configmap) ?

**Reply by nishant-vvdn** [reply]
If I'm understanding correctly you are saying that we can add 

> emqxAclConfig: >
>   %% I want to disable 'dashboard' user.
>   %% {allow, {user, "dashboard"}, subscribe, ["$SYS/#"]}.
>   {allow, {ipaddr, "127.0.0.1"}, pubsub, ["$SYS/#", "#"]}.
>   {deny, all, subscribe, ["$SYS/#", {eq, "#"}]}.
>   {allow, all}.

  
  in the configmap and then EMQX will itself pick this up for acl.conf ?

**Reply by zmstone** [reply]
No, emqx will not query k8s for configs,
Rather, you need to make use of the configmap in your k8s deployment or statefulset
to overwrite the acl.conf shipped in the docker image.

**Reply by zmstone** [reply]
example: https://github.com/emqx/emqx/blob/a48e7df4f5900dfb477c0423579c60d0ac6c03f4/deploy/charts/emqx/templates/StatefulSet.yaml#L64-L69

**Reply by nishant-vvdn** [reply]
I will get back to you after I try this.

Thanks, @zmstone
